backend/src/middleware/userInteractions.js

```diff
@@ -0,0 +1,44 @@
+const createRelatedCypher = (relation) => `
+MATCH (user:User { id: $currentUser})
+MATCH (post:Post { id: $postId})
+OPTIONAL MATCH (post)<-[r:${relation}]-(u:User)
+WHERE NOT u.disabled AND NOT u.deleted
+WITH user, post, count(DISTINCT u) AS count
+MERGE (user)-[relation:${relation} { }]->(post)
+ON CREATE
+SET relation.count = 1,
```

Comment:
I do not get why you would set a count on the relation?! Isn't the relation between a specific user & a specific post, is it?